src/components/TransactionStatusToast/TransactionStatusToast.tsx

```diff
@@ -0,0 +1,130 @@
+import { addNewCustomToast, deleteCustomToast } from '../../utils';
+import { CustomToast } from '../../UI';
+import React, { useEffect, useMemo } from 'react';
+import { CustomToastType } from '../../types/toasts';
+import { useSelector } from '../../reduxStore/DappProviderContext';
+import { customToastsSelector } from '../../reduxStore/selectors';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faInfo, faTimes, faWarning } from '@fortawesome/free-solid-svg-icons';
+import globalStyles from 'assets/sass/main.scss';
+import classNames from 'classnames';
+import { TRANSACTION_STATUS_TOAST_ID } from 'constants/index';
+import styles from './transaction-status-toast.scss';
+
+type ErrorToastProps = {
+  type?: 'info' | 'error' | 'warning';
+  message: string;
+  duration?: number;
+  show?: boolean;
+  onDelete?: () => void;
+};
+
+export const TransactionStatusToast: React.FC<ErrorToastProps> = ({
+  show,
+  message,
+  duration = 30000,
+  onDelete,
+  type = 'info'
+}) => {
+  const customToasts = useSelector(customToastsSelector);
+
+  const errorToast = useMemo(
+    () =>
+      customToasts.find(
+        (customToast: CustomToastType) =>
+          customToast.toastId === TRANSACTION_STATUS_TOAST_ID
+      ),
+    [customToasts]
+  );
+
+  const showErrorToast = () => {
+    addNewCustomToast({
+      toastId: TRANSACTION_STATUS_TOAST_ID,
+      message,
+      type: 'custom'
+    });
+  };
+
+  const handleDelete = () => {
+    if (Boolean(errorToast)) {
+      deleteCustomToast(TRANSACTION_STATUS_TOAST_ID);
+      onDelete?.();
+    } else {
+      deleteCustomToast(TRANSACTION_STATUS_TOAST_ID);
+    }
+  };
+
+  useEffect(() => {
+    if (show) {
+      showErrorToast();
+    }
+  }, [show]);
+
+  const StatusIcon = useMemo(() => {
```

Comment:
What if we lift this up and use a renderProp or pass the Icon as a child to the Toast, to avoid rerendering this when something changes and avoid using useMemo? We can create a component called StatusIcon that receives a type and renders all this